lib/configure.js

```diff
@@ -127,6 +128,18 @@ function configure (gyp, argv, callback) {
 
     // set the target_arch variable
     variables.target_arch = gyp.opts.arch || process.arch || 'ia32'
+    log.verbose('build/' + configFilename,
+      'target arch: ' + variables.target_arch)
+
+    // set the target_platform variable
+    variables.target_platform = process.platform
```

Comment:
Maybe we should allow manual override here as well? What about cross-platform builds?